### PR TITLE
feat(bridge): allow sell=buy for native/wrapped

### PR DESCRIPTION
**Pull request by shoom3301**
Since the backend supports swapping native/wrapped to native/wrapped, we can enable it in the SDK.

1. Removed the restriction (https://github.com/cowprotocol/cow-sdk/commit/d64394c92c79ac957766c43b4ca1c0a14530edfb)
2. Added more info to PRIORITY_STABLECOIN_TOKENS. I will use it in the next step (https://github.com/cowprotocol/cow-sdk/commit/ada3db9c48879e4935c12d0a4f3b6bd36fc376db)
3. Gave the highest priority to the case when intermediate = destination. I only implemented the simplest way to compare intermediate and destination tokens - using `PRIORITY_STABLECOIN_TOKENS` and the token symbol. (https://github.com/cowprotocol/cow-sdk/commit/8d1d24f05829ee590475210b03ce588ecdd195ef)
4. Renamed `TokenPriority` enum to avoid abstract values

I would keep `allowIntermediateEqSellToken` flag for awhile in case if we will need to disable the feature.

## Summary by CodeRabbit

* **Refactor**
  * Improved intermediate-token selection for bridging: destination stablecoin matches now take highest precedence, followed by sell-token, registry, correlated, and native-token fallbacks — yielding more optimal cross-chain routing.
  * Added a new top priority level to prefer explicit destination stablecoins.

* **Bug Fixes**
  * Resilient fallback when correlated-token lookup fails; respects allow-equal-sell-token setting.

* **Tests**
  * Expanded and reorganized tests to cover new priority behavior and edge cases.

### CODE REVIEW

File: packages/bridging/src/BridgingSdk/determineIntermediateToken.ts [hunk 1]
```diff
@@ -1,17 +1,27 @@
...
  */
 enum TokenPriority {
-  HIGHEST = 5, // Same as sell token
-  HIGH = 4, // USDC/USDT from hardcoded registry
-  MEDIUM = 3, // Tokens in CMS correlated tokens list
-  LOW = 2, // Blockchain native token
-  LOWEST = 1, // Other tokens
+  STABLECOIN_MATCHES_DESTINATION = 6, // Stable coin, same as destination token
+  MATCHES_SELL = 5, // Same as sell token
+  STABLECOIN = 4, // USDC/USDT from hardcoded registry
```
Comment (alfetopito) [inline packages/bridging/src/BridgingSdk/determineIntermediateToken.ts]: It's a bit confusing to have stables on 2 different levels IMO

File: packages/bridging/src/BridgingSdk/determineIntermediateToken.ts [hunk 2]
```diff
@@ -48,35 +62,41 @@ export async function determineIntermediateToken(
...
-    const isNativeOrWrapped = isNativeToken(token) || isWrappedNativeToken(token)
+    const isStableCoin = isStablecoinPriorityToken(token.chainId, token.address)
 
-    if (areAddressesEqual(token.address, sourceTokenAddress)) {
-      // Native/wrapped -> native/wrapped is not supported yet (backend restriction)
-      if (!(isSellNativeOrWrapped && isNativeOrWrapped)) {
-        return { token, priority: TokenPriority.HIGHEST }
+    if (destinationStableCoin && isStableCoin) {
+      const matchesDestinationTokenSymbol =
+        !!token.symbol && token.symbol.toLowerCase() === destinationStableCoin?.symbol?.toLowerCase()
```
Comment (alfetopito) [inline packages/bridging/src/BridgingSdk/determineIntermediateToken.ts:79]: That's not very stable, is it?
There are several flavours of USDC for example, and they can have slightly different symbols:
- USDC
- USDCe
- USDC.e

Same for USDT (USDT0 on Plasma) and so on.

Didn't you propose to do a map of known addresses per chain?

Comment (shoom3301) [reply packages/bridging/src/BridgingSdk/determineIntermediateToken.ts:79]: >That's not very stable, is it?

Correct. We use `PRIORITY_STABLECOIN_TOKENS` map for that.
The idea is to have an exact match here. Since the map is hardcoded, we can quite safely compare the tokens using symbol.
For a case `USDC => USDC.e` we have `TokenPriority.STABLECOIN`

File: packages/bridging/src/BridgingSdk/determineIntermediateToken.ts (hunk 1 above)
Comment (shoom3301) [reply packages/bridging/src/BridgingSdk/determineIntermediateToken.ts]: https://github.com/cowprotocol/cow-sdk/pull/897#discussion_r3402233418

Maybe the enum naming is not really good, but I have no better ideas

File: packages/bridging/src/BridgingSdk/determineIntermediateToken.ts (hunk 2 above)
Comment (Danziger) [reply packages/bridging/src/BridgingSdk/determineIntermediateToken.ts:79]: Agree with this, but given it is hardcoded, I would just add a comment in the code.

**Review by Danziger — APPROVED** [review]

File: packages/bridging/src/BridgingSdk/determineIntermediateToken.ts (hunk 2 above)
Comment (shoom3301) [reply packages/bridging/src/BridgingSdk/determineIntermediateToken.ts:79]: Thanks!
Improved the comments:
https://github.com/cowprotocol/cow-sdk/pull/897/commits/6ae399963957e70e5182078996c347c04a816238
